### ISSUE TITLE
Autocomplete - does not hide() on click of item

### ISSUE THREAD

**Issue by rosscdh** [angular/material#1681]
Hi there, firstly thanks for the amazing lib really great!

I'm having an issue using the autocomplete, it works in the sense that it displays ok (except the does not popopen on recieving data but rather on next keyup.. which is strange too)..

but the actual issue i am having is that when I click on an item; the dropdown does not hide(). 

any thought would be appreciated.

```
  <md-autocomplete
      md-min-length="{{minLength}}"
      md-search-text="data.searchTerm"
      md-selected-item="selectedItem"
      md-search-text-change="startAutosuggest(data.searchTerm)"
      md-items="item in data.autosuggestResults"
      md-item-text="item.text"
      placeholder="Such Deine Produkte">
    <span md-highlight-text="data.searchTerm">{{ item.text }}</span>
  </md-autocomplete>
```

**Comment by RomainFleury** [comment]
Maybe add an optional  "md-select-action" parameter, fired with selectedItem as first parameter when select is called in directive.
=> autocompleteController.js @ 192
```javascript
function select (index) {
      $scope.selectedItem = self.matches[index];
      $scope.searchText = getDisplayValue($scope.selectedItem) || $scope.searchText;

      if(typeof($scope.selectAction) === 'function'){
          $scope.selectAction($scope.selectedItem);
      }

      self.hidden = true;
      self.index = -1;
      self.matches = [];
    }
```

?

**Comment by kuhnroyal** [comment]
Try to use `md-selected-item-change`

**Comment by kendrickw** [comment]
@rosscdh, not sure if you are experiencing this problem: #1974

**Comment by robertmesserle** [comment]
This seems to have been resolved in master - most-likely by this commit 5f14126954fa10911e00a8c18bf8a070133ae5de.
